appserver/admingui/payara-console-extras/src/main/java/fish/payara/admingui/extras/rest/PayaraRestApiHandlers.java

```diff
@@ -166,6 +168,95 @@ public static void sendAsadminCommandToSelectedInstances(HandlerContext handlerC
     }
     
     /**
+     * Gets the REST endpoints from a given app name and optional component name
+     * @param handlerCtx 
+     */
+    @Handler(id = "py.getRestEndpoints",
+        input = {
+            @HandlerInput(name = "appName", type = String.class, required = true),
+            @HandlerInput(name = "componentName", type = String.class, required = false)},
+        output = {
+            @HandlerOutput(name = "result", type = java.util.List.class)})
+    public static void getRestEndpoints(HandlerContext handlerCtx) {
+        List result = new ArrayList();
+        try{
+            String appName = (String) handlerCtx.getInputValue("appName");
+            String encodedAppName = URLEncoder.encode(appName, "UTF-8");
+            String componentName = (String) handlerCtx.getInputValue("componentName");
+            String encodedComponentName = URLEncoder.encode(componentName, "UTF-8");
+            String prefix = GuiUtil.getSessionValue("REST_URL") + "/applications/application/" + encodedAppName;
+            
+            // get the extra properties from the list-rest-endpoints command, passing in the component name
+            Map attrMap = new HashMap();
+            attrMap.put("componentname", encodedComponentName);
+            Map payaraEndpointDataMap = RestUtil.restRequest(prefix + "/list-rest-endpoints", attrMap, "GET", null, false, false);
+            Map payaraEndpointsExtraProps = (Map) ((Map) ((Map) payaraEndpointDataMap.get("data")).get("extraProperties"));
+
+            // Check if the command returned any endpoints
+            if((Map)payaraEndpointsExtraProps.get("endpointMap") != null) {
+                Map<String, String> endpointMap = (Map)payaraEndpointsExtraProps.get("endpointMap");
+                for(String key : endpointMap.keySet()) {
+                    // Count through returned endpoints
+                    Map<String, String> rowMap = new HashMap<>();
```

Comment:
Must be 
Map<String, String> rowMap = new HashMap<>(String, String);
in order to compile with JDK7